dlrover/trainer/tensorflow/util/reader_util.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dlrover.trainer.tensorflow.util.common_util import singleton
+from dlrover.trainer.util.log_util import default_logger as logger
+
+
+@singleton
+class ReaderRegistry:
+    """ """
+
+    def __init__(self):
+        self.reader_registry = {}
+
+    def register_reader(self, reader_name, reader_class):
```

Comment:
`dataset_name` is better than `reader_name` tell users dlrover trainer use the reader to read which dataset.